packages/next-app/src/components/Sidebar.tsx

```diff
@@ -0,0 +1,84 @@
+import { MagnifyingGlassIcon } from "@radix-ui/react-icons";
+import Link from "next/link";
+import { useState } from "react";
+
+import { Actions } from "#/app/daoadmin/(components)/Actions";
+import { ActionAttribute, useAdminTools } from "#/contexts/AdminToolsContext";
+import { hardcodedData } from "#/utils/hardcodedData";
+
+import { FilterDropdown, IFilter } from "./FilterDropdown";
+
+export function Sidebar() {
+  const data = hardcodedData;
+  const [querySearch, setQuerySearch] = useState("");
+  const { selectedAction, handleSetAction, selectedFilters } = useAdminTools();
+
+  const handleSearchChange = (e: React.FormEvent<HTMLInputElement>) => {
+    setQuerySearch(e.currentTarget.value);
+  };
```

Comment:
Is there any reason to have this living here instead of centralizing this filtering related logic to the Admin context? 

IMO this absolutely must not live here, and I think it only strengthens the case of creating a separate context for filtering. Also `#would` consider if there's any reason to put the search query on the URL here. What's the value of putting the search query in the URL? Would users share this URL with other folks? If not, would consider not having this at all to avoid complicating things.

Reply:
Already removed from components 😃 

Not quite sure what you mean by 'search query on the URL'. This is just a search query to navigate through the DAO Actions. Could you clarify what you meant? 🙂